base/linalg/blas.jl

```diff
@@ -463,6 +463,55 @@ function axpy!(alpha::Number, x::Array{T}, rx::Union{UnitRange{Ti},Range{Ti}},
     y
 end
 
+"""
+    axpby!(a, X, b, Y)
+
+Overwrite `Y` with `a*X + b*Y`, where `a` and `b` are scalars. Return `Y`.
+
+# Examples
+```jldoctest
+julia> x = [1., 2, 3];
+
+julia> y = [4., 5, 6];
+
+julia> Base.BLAS.axpby!(2., x, 3., y)
+3-element Array{Float64,1}:
+14.0
+19.0
+24.0
+```
+"""
+function axpby! end
+
+for (fname, elty) in ((:daxpby_,:Float64),
+                  (:saxpby_,:Float32),
+                  (:zaxpby_,:Complex128),
+                  (:caxpby_,:Complex64))
+@eval begin
+            # SUBROUTINE DAXPBY(N,DA,DX,INCX,DB,DY,INCY)
+            # DY <- DA*DX + DB*DY
+            #*     .. Scalar Arguments ..
+            #      DOUBLE PRECISION DA,DB
+            #      INTEGER INCX,INCY,N
+            #*     .. Array Arguments ..
+            #      DOUBLE PRECISION DX(*),DY(*)
+    function axpby!(n::Integer, alpha::($elty), dx::Union{Ptr{$elty}, DenseArray{$elty}}, incx::Integer, beta::($elty), dy::Union{Ptr{$elty}, DenseArray{$elty}}, incy::Integer)
```

Comment:
getting a bit beyond the line length recommendation

would also be more conventional to indent the whole `@eval` block since it's inside a for loop (or do `@eval function` since the `begin` is somewhat superfluous)